### PR TITLE
BUGFIX: ONE-3017 allow getting deprecated objects

### PR DESCRIPTION
**Pull request by ZdenekVidensky**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/288/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/255/ : SUCCESS in 1m 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/293/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/261/ : SUCCESS in 1m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/294/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/262/ : SUCCESS in 1m 44s

File: src/metadata.ts [hunk 1]
```diff
@@ -97,7 +98,7 @@ export class MetadataModule {
         };
 
         const uri = `/gdc/md/${projectId}/objects/query`;
-        const query = pick({ limit: 50, ...options }, ['category', 'mode', 'author', 'limit']);
+        const query = pick({ limit: 50, ...options }, ['category', 'mode', 'author', 'limit', 'deprecated']);
```
Comment (pbenes) [inline src/metadata.ts]: check if we could have at least one test for param propagation

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/295/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/263/ : SUCCESS in 1m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/296/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/264/ : SUCCESS in 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/297/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/265/ : SUCCESS in 54s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/63/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/266/ : SUCCESS in 46s

**Pull request merged**